### PR TITLE
Unit Test Shadows for Storage and tests for Case Index Removal

### PR DESCRIPTION
**Pull request by ctsims**
X-requested with: https://github.com/dimagi/commcare/pull/53

Added shadows for IO-related classes to test framework (Application and SqlStorage can be instantiated)

Wired in configs between robolectric and core project.

Configured Resources to be available at the classpath level for unit tests both in ANT and local runtime.

Made minor constructor changes for some DB related classes to let them not pull from the static context (nothing should be doing that anyway, ideally!!!)

Added tests and a fix for case transactions which remove indexes

### CODE REVIEW

File: unit-tests/src/org/commcare/android/shadows/SQLiteDatabaseNative.java [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+@Implements(net.sqlcipher.database.SQLiteDatabase.class)
+public class SQLiteDatabaseNative {
+    private android.database.sqlite.SQLiteDatabase db;
+    
+    public void __constructor__(String path, char[] password, CursorFactory factory, int flags) {
+        db = android.database.sqlite.SQLiteDatabase.openDatabase(path, null, flags);
+    }
+    
+    public void __constructor__(String path, char[] password, CursorFactory factory, int flags, SQLiteDatabaseHook hook) {
+        db = android.database.sqlite.SQLiteDatabase.openDatabase(path, null, flags);
```
Comment (orangejenny) [inline unit-tests/src/org/commcare/android/shadows/SQLiteDatabaseNative.java]: Why is there a constructor that takes a hook if it doesn't do anything with it?

Comment (orangejenny) [inline unit-tests/src/org/commcare/android/shadows/SQLiteDatabaseNative.java]: Oh, nm, this is implementing SQLiteDatabase so presumably there's no choice.

File: unit-tests/src/org/commcare/android/tests/processing/ProcessingTest.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        assertEquals("Changed Type", "changed_unit_test", c2.getTypeId());
+    }
+
+
+    private TransactionParserFactory getFactory(final SQLiteDatabase db) {
+        return new TransactionParserFactory() {
+            
+            public TransactionParser getParser(String name, String namespace, KXmlParser parser) {
+                if(LedgerXmlParsers.STOCK_XML_NAMESPACE.equals(namespace)) {
+                    //return new LedgerXmlParsers(parser, CommCareApplication._().getUserStorage(Ledger.STORAGE_KEY, Ledger.class));
```
Comment (orangejenny) [inline unit-tests/src/org/commcare/android/tests/processing/ProcessingTest.java]: Deliberately commented out?

File: unit-tests/src/org/commcare/android/shadows/SQLiteDatabaseNative.java (hunk 1 above)
Comment (ctsims) [inline unit-tests/src/org/commcare/android/shadows/SQLiteDatabaseNative.java]: Yeah, it's a potential issue to check for in the future, but the incoming Hook here is the sqlcipher type, so we can't forward it. We'd have to make another wrapper like we did for the cursor, which is a super pain due to the way these SQL classes do a bunch of shit in their constructors.

File: unit-tests/src/org/commcare/android/tests/processing/ProcessingTest.java (hunk 2 above)
Comment (ctsims) [inline unit-tests/src/org/commcare/android/tests/processing/ProcessingTest.java]: I'll clean this up. Agree that this isn't a good state to leave things in.

**Comment by ctsims** [comment]
@orangejenny Fixed up the rest of the issues you mentioned

**Pull request merged**